data/secscan_model/secscan_v4_model.py

```diff
@@ -68,25 +84,29 @@ def load_security_information(self, manifest_or_legacy_image, include_vulnerabil
         if status.index_status == IndexStatus.FAILED:
             return SecurityInformationLookupResult.with_status(ScanLookupStatus.FAILED_TO_INDEX)
 
-        if status.index_status == IndexStatus.UNSUPPORTED:
+        if status.index_status == IndexStatus.MANIFEST_UNSUPPORTED:
             return SecurityInformationLookupResult.with_status(
                 ScanLookupStatus.UNSUPPORTED_FOR_INDEXING
             )
 
-        if status.index_status in [
-            IndexStatus.TIMED_OUT,
-            IndexStatus.WAITING,
-            IndexStatus.IN_PROGRESS,
-        ]:
+        if status.index_status == IndexStatus.IN_PROGRESS:
             return SecurityInformationLookupResult.with_status(ScanLookupStatus.NOT_YET_INDEXED)
 
         assert status.index_status == IndexStatus.COMPLETED
 
-        # TODO(alecmerdler): Fetch data from Clair API
-        return SecurityInformationLookupResult.with_status(ScanLookupStatus.NOT_YET_INDEXED)
+        report = self._secscan_api.vulnerability_report(manifest_or_legacy_image.digest)
+
+        # TODO(alecmerdler): Should we also return this status if `state != status.indexer_hash`...?
```

Comment:
No, as the intention is we return the older indexed information until such time as the manifest gets re-indexed; this will ensure that we can reindex "in the background".

Reply:
Should we provide a field that indicates that the current scan is outdated?

Reply:
That's a really good idea

Reply:
Let's address this in a follow up PR.

Reply:
Add a TODO then